pkg/controller/storagecluster/reconcile.go

```diff
@@ -664,18 +664,19 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
 
 // determineFailureDomain determines the appropriate Ceph failure domain based
 // on the storage cluster's topology map
-func determineFailureDomain(topologyMap *ocsv1.NodeTopologyMap) string {
+func determineFailureDomain(sc *ocsv1.StorageCluster) string {
+	if sc.Status.FailureDomain != "" {
+		return sc.Status.FailureDomain
+	}
+	topologyMap := sc.Status.NodeTopologies
 	failureDomain := "rack"
-
 	for label, labelValues := range topologyMap.Labels {
 		if strings.Contains(label, "zone") {
 			if len(labelValues) >= 3 {
 				failureDomain = "zone"
-				break
 			}
 		}
 	}
-
```

Comment:
Just a taste thing (and I know that e.g. @jarrpa is not as focused on these as I am): I personally do like some new lines to group the code more for my eyes, but more importantly, I don't like changes that are not related to the actual change to be sneaked in, even if they are just whitespace changes.

So my rule of thumb would be: if you are changing this context anyway, feel free to do whitespace or formatting changes in the same commit, but if there are no changes anyway in this context the commit, keep the diff smaller. If you still want to do the whitespace or formatting changes, do them as separate commits maybe even in the same PR.